src/ebuckets.c

```diff
@@ -1900,6 +1902,7 @@ void ebDefragRaxBucket(EbucketsType *type, raxIterator *ri,
         if (newSegHdr) {
             /* Update next segment's prev to point to the defragmented segment. */
             nextSegHdr->prevSeg = newSegHdr;
+            nextSegHdr->firstSeg = (FirstSegHdr *)firstSegHdr;
```

Comment:
don't we need to update that on all segments?  (not just when newSegHdr is non NULL)
i.e. don't depend on `newSegHdr`, but rather add a var newFirstSegHdr? or just update unconditionally?